RadarSDK/RadarLogBuffer.m

```diff
@@ -0,0 +1,74 @@
+//
+//  RadarLogBuffer.m
+//  RadarSDK
+//
+//  Copyright © 2021 Radar Labs, Inc. All rights reserved.
+//
+
+#import "Radar.h"
+#import "RadarLog.h"
+#import "RadarLogBuffer.h"
+
+static const int MAX_BUFFER_SIZE = 1000;
+static const int PURGE_AMOUNT = 500;
+
+static NSString *const kPurgedLogLine = @"----purged oldest logs! ----";
+
+@implementation RadarLogBuffer {
+    NSMutableArray<RadarLog *> *mutableLogBuffer; // define log buffer
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        mutableLogBuffer = [NSMutableArray<RadarLog *> new]; // initialize log buffer
+    }
+    return self;
+}
+
++ (instancetype)sharedInstance {
+    static dispatch_once_t once;
+    static id sharedInstance;
+    dispatch_once(&once, ^{
+        sharedInstance = [self new];
+    });
+    return sharedInstance;
+}
+
+- (void)write:(RadarLogLevel)level message:(NSString *)message {
+    // purge oldest if reached the max buffer size
+    NSUInteger logLength = [mutableLogBuffer count];
+    if (logLength >= MAX_BUFFER_SIZE) {
+        [self purgeOldestLogs];
+    }
+    // add new log to buffer
+    RadarLog *radarLog = [[RadarLog alloc] initWithMessage:message level:level];
+    [mutableLogBuffer addObject:radarLog];
```

Comment:
Nit: This function takes a `RadarLogLevel` and an `NSString`, but the `RadarLog`'s initializer takes an `NSString` and a `RadarLogLevel`. I would swap the parameters of the `RadarLog`'s initializer.